lib/elasticsearch.ts

```diff
@@ -70,14 +78,108 @@ function filter_old_indices(current_indices) {
+    request_es_no_body("del", `/${index}`).then(() => resolve(index)).catch(reject);
   });
 }
 
 function delete_indices(indices) {
-  return Promise.all(_.map(indices, (index) => delete_index(index)));
+  return Promise.all(_.map(indices, delete_index));
 }
 
 export function clear_old_indices() {
   return get_indices().then(filter_old_indices).then(delete_indices);
 }
+
+function get_aliases() {
+  return new Promise((resolve, reject) => {
+    get_es("/_aliases").then((indices) => {
+      // The aliases endpoint returns a map from index name to metadata about those indexes.
+      // e.g.
+      //   {
+      //     ".kibana-4": {
+      //       "aliases": {}
+      //     },
+      //     "logs-2016.04.02": {
+      //       "aliases": {
+      //         "last_2days": {},
+      //         "last_2weeks": {},
+      //         "last_day": {},
+      //         "last_week": {}
+      //       }
+      //     },
+      //   }
+      // It's much easier to work with a map from alias name to list of indexes in the alias.
+      const aliases = {};
+      for (const index of Object.keys(indices)) {
+        // Filter out the kibana index
+        if (index.indexOf(".kibana") !== -1) {
+          continue;
+        }
+        for (const alias of Object.keys(indices[index].aliases)) {
+          if (!aliases[alias]) {
+            aliases[alias] = [];
+          }
+          aliases[alias].push(index);
+          // Keep the results sorted and unique
+          aliases[alias] = _.chain(aliases[alias]).sortBy(_.identity).uniq(true).value();
+        }
+      }
+
+      resolve(aliases);
+    }).catch((err) => {
+      reject(err);
+    });
+  });
+}
+
+function remove_unmanaged_aliases(aliases) {
+  return new Promise((resolve) => {
+    for (const alias of Object.keys(aliases)) {
+      if (!config.aliases.mappings[alias]) {
+        delete aliases[alias];
+      }
+    }
+    resolve(aliases);
+  });
+}
+
+function update_alias_state(current_indices, alias) {
+  return new Promise((resolve, reject) => {
+    const acceptable_indices = [];
+    let today = moment();
+    for (let i = 0; i < config.aliases.mappings[alias]; i++) {
+      acceptable_indices.push(`${config.indices.prefix}-${today.format("YYYY.MM.DD")}`);
+      today = today.subtract(1, "days");
+    }
+    const indices_to_remove = _.difference(current_indices, acceptable_indices);
+    const indices_to_add = _.difference(acceptable_indices, current_indices);
+
+    if (!indices_to_remove.length && !indices_to_add.length) {
+      resolve();
+      return;
+    }
+
+    const action_el = (action, index) => {
+      const out = {};
+      out[action] = {index, alias};
+      return out;
+    };
+    const actions = _.map(indices_to_remove, _.partial(action_el, "remove"))
+      .concat(_.map(indices_to_add, _.partial(action_el, "add")));
+    request_es("post", "/_aliases", {actions}).then(() => {
+      resolve();
+    }).catch((err) => {
```

Comment:
these two blocks could probably use some comments to explain the format needed for posting to `/_aliases`